components/files/file.jsx

```diff
@@ -0,0 +1,216 @@
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { FILES_FILE } from '../../utilities/constants';
+
+import FileFigure from './private/file-figure';
+import FileActions from './private/file-actions';
+
+const displayName = FILES_FILE;
+
+const propTypes = {
+	/**
+	 * **Assistive text for accessibility**
+	 *  * download - description for the download button if present
+	 *  * image - description for the file image
+	 *  * link - description for the file link
+	 *  * loading - description for the loading spinner if present
+	 *  * moreActions - description for the more actions dropdown if present
+	 */
+	assistiveText: PropTypes.shape({
+		download: PropTypes.string,
+		image: PropTypes.string,
+		link: PropTypes.string,
+		loading: PropTypes.string,
+		moreActions: PropTypes.string,
+	}),
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * Controls different cropping aspect ratios for the component
+	 */
+	crop: PropTypes.oneOf(['16-by-9', '4-by-3', '1-by-1']),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Action to be done on clicking download button; doesn't show download button if empty
+	 */
+	onClickDownload: PropTypes.func,
+	/**
+	 * Function that is called when image is clicked; can be used instead of href for more advanced event handling
+	 */
+	onClickImage: PropTypes.func,
+	/**
+	 * Dropdown for more actions button; doesn't show more actions button if empty
+	 */
+	moreActionsDropdown: PropTypes.node,
+	/**
+	 * Icon associated with the file. Accepts an Icon component
+	 */
+	icon: PropTypes.node,
+	/**
+	 * Icon to be shown in top left corner of File component. Accepts an Icon component
+	 */
+	externalIcon: PropTypes.node,
+	/**
+	 * Link to thumbnail image
+	 */
+	image: PropTypes.string,
+	/**
+	 * Controls whether file preview is loading
+	 */
+	isLoading: PropTypes.bool,
+	/**
+	 * Href attribute for image
+	 */
+	href: PropTypes.string,
+	/**
+	 * Labels for the File Component
+	 * * image - title for the file. Required.
+	 */
+	labels: PropTypes.shape({
+		title: PropTypes.string.isRequired,
+	}),
+	/**
+	 *  Controls whether the file's title should be visible
+	 */
+	hasNoVisibleTitle: PropTypes.bool,
+};
+
+const defaultProps = {
+	assistiveText: {
+		download: 'download',
+		link: 'Preview:',
+		loading: 'loading',
+		moreActions: 'more actions',
+	},
+	crop: '16-by-9',
+	href: 'javascript:void(0);',
+	isLoading: false,
+	hasNoVisibleTitle: false,
+};
+/**
+ * File is a component that represents content uploaded as an attachment.
+ */
+class File extends React.Component {
+	static injectMoreActionsStyles() {
+		return (
+			<style>{`
+					.dsr-file__more-actions-dropdown  ul.dropdown__list li.slds-dropdown__item > a:before
```

Comment:
@kevinparkerson Did we follow up with SLDS about this bug?

Reply:
@interactivellama was going to post about it to GUS real soon :)